### PR TITLE
Expose the metadata socket option from libzmq

### PR DESCRIPTION
**Pull request by scottcannonjr**

### CODE REVIEW

File: src/zmqpp/socket_options.hpp [hunk 1]
```diff
@@ -116,6 +116,7 @@ ZMQPP_COMPARABLE_ENUM socket_option {
 	heartbeat_timeout         = ZMQ_HEARTBEAT_TIMEOUT, /*< ZMPT heartbeat timeout - set only */
 	heartbeat_ttl             = ZMQ_HEARTBEAT_TTL, /*< ZMPT heartbeat interval - set only */
 	invert_matching           = ZMQ_INVERT_MATCHING, /*< ZMPT invert state for PUB/SUB message filters */
+  metadata                  = ZMQ_METADATA, /*< socket metadata key-value pairs */
```
Comment (scottcannonjr) [inline src/zmqpp/socket_options.hpp]: I guess I'll have to break down and insert some TAB characters here, much as that makes me cringe...

Comment (smanders) [reply src/zmqpp/socket_options.hpp]: haha, yes... I cringe when I edit a third-party library for things like tabs, too -- or ALL_CAPS CMAKE CALLS

**Review by smanders — APPROVED** [review]
looks good, to me!

**Comment by smanders** [comment]
looks like I've made one other commit to the xp21.07.09 branch since you branched off and did this work
https://github.com/smanders/zmqpp/commits/xp21.07.09

but the one commit shouldn't have any impact on what you're doing and your changes will merge just fine
https://github.com/smanders/zmqpp/commit/48adfc92825231775753c7ab34b35f86b91ebd51

**Pull request merged**